### PR TITLE
feat(hugging_face): Add support for multiple models and dimensions

### PR DESCRIPTION
**Pull request by bricesanchez**
Hello! Thank you for this awesome gem!

I'm currently playing with it and i've seen Hugging face ruby gem can now support custom embedding models:

see: https://github.com/alchaplinsky/hugging-face/commit/bee85a443bd085acbc61ad8926afa574aadfc7a4

I've added the ability to change the embedding model and did some clean up in the class.

### CODE REVIEW

**Comment by andreibondarev** [comment]
@bricesanchez Thank you for the PR, looks like the specs are failing.

**Comment by bricesanchez** [comment]
@andreibondarev You're right!

It looks like elasticsearch class looks like for the LLM dimensions (lib/langchain/vectorsearch/elasticsearch.rb:107). 
And with my change, it breaks.

But it looks like there is no `dimension` mention in the hugging_face gem. 

I can re-add it for simplicity.

What do you think?

**Comment by andreibondarev** [comment]
@bricesanchez Take a look at how we do it in the Ollama integration: https://github.com/patterns-ai-core/langchainrb/blob/b3a1b65c4a9dffc5ce33eee8410551002adb9357/lib/langchain/llm/ollama.rb#L53-L59

File: lib/langchain/llm/hugging_face.rb [hunk 1]
```diff
@@ -24,10 +21,21 @@ class HuggingFace < Base
...
       @client = ::HuggingFace::InferenceApi.new(api_token: api_key)
+      @defaults = DEFAULTS.merge(default_options)
+    end
+
+    # Returns the # of vector dimensions for the embeddings
+    # @return [Integer] The # of vector dimensions
+    def default_dimensions
+      # since Huggin Face can run multiple models, look it up or generate an embedding and return the size
+      @default_dimensions ||= @defaults[:dimensions] ||
+        EMBEDDING_SIZES.fetch(defaults[:embeddings_model_name].to_sym) do
```
Comment (andreibondarev) [inline lib/langchain/llm/hugging_face.rb:36]: I think there's a bug here. See the execution below:
```ruby
hf = Langchain::LLM::HuggingFace.new(api_key: ENV["HUGGING_FACE_API_KEY"], default_options: {embeddings_mod
el_name: "mixedbread-ai/mxbai-embed-large-v1"})
hf.default_dimensions
#=> 384
```
even though [this](https://huggingface.co/mixedbread-ai/mxbai-embed-large-v1) model is 512 by default.

**Comment by andreibondarev** [comment]
@bricesanchez Thank you! Left you a comment.

File: lib/langchain/llm/hugging_face.rb (hunk 1 above)
Comment (bricesanchez) [reply lib/langchain/llm/hugging_face.rb]: Yes sorry i fixed it. 

FYI, it looks like the default dimension for `mixedbread-ai/mxbai-embed-large-v1` is `1024`: https://www.mixedbread.ai/docs/embeddings/mxbai-embed-large-v1#model-description

**Comment by andreibondarev** [comment]
@bricesanchez I'm going to review it shortly.

**Review by andreibondarev — APPROVED** [review]
@bricesanchez Thank you for the PR! Looks great!

**Comment by bricesanchez** [comment]
Thank you @andreibondarev!

**Pull request merged**